src/websocket/stream.py

```diff
@@ -115,57 +99,51 @@ async def echo(self, websocket, path) -> None:
-                        await websocket.send("control message unknown")
-
-            except websockets.exceptions.ConnectionClosedOK:
-                self.logger.debug("Client left")
-                self.close_stream = True
-            except websockets.exceptions.ConnectionClosedError:
-                self.logger.debug("Client left unexpectedly")
-                self.close_stream = True
-            except Exception:
-                self.logger.error(
-                    "Error while receiving message: {}".format(traceback.format_exc())
-                )
-                self.close_stream = True
+                        await websocket.send_text("control message unknown")
+
+        except WebSocketDisconnect:
+            self.logger.debug("Client disconnected")
+            self.close_stream = True
+        except Exception:
+            self.logger.error(
+                f"Error while receiving message: {traceback.format_exc()}"
+            )
+            self.close_stream = True
+        finally:
+            await websocket.close()
 
     async def transcribe_all_chunk_cache(
-        self, websocket, chunk_cache, recent_cache, skip_send=False
-    ) -> str:
+        self, websocket: WebSocket, chunk_cache, recent_cache, skip_send=False
+    ) -> None:
         """Function to transcribe a chunk of audio"""
         try:
             start_time = time.time()
-            result: str = ""
             bytes_to_transcribe = self.last_final_bytes + chunk_cache
             data = self.transcriber._transcribe(
                 bytes_to_transcribe,
                 "Beginning of transcription:" + self.last_final_result_object["text"],
             )
             self.adjust_threshold_on_latency()
-            # we want the time when the final started, so it is the self.overall_transcribed_bytes time minus the recently added bytes
             overall_transcribed_seconds = (
                 len(self.overall_transcribed_bytes) / BYTES_PER_SECOND
             )
-            # We need to remove the time of the repeatedly transcribed bytes from the overall transcribed seconds
             last_final_bytes_seconds = len(self.last_final_bytes) / BYTES_PER_SECOND
             overall_transcribed_seconds -= last_final_bytes_seconds
-
             self.overall_transcribed_bytes += recent_cache
 
+            result = {"result": [], "text": ""}
             if "segments" in data:
                 words = []
                 for segment in data["segments"]:
                     for word in segment["words"]:
-                        # make float with 6 digits after point
-                        start = float("{:.6f}".format(float(word["start"])))
-                        end = float("{:.6f}".format(float(word["end"])))
-                        conf = float("{:.6f}".format(float(word["probability"])))
-                        word = word["word"].strip()
+                        start = float(f"{float(word['start']):.6f}")
```

Comment:
we do a cast str -> float -> str -> float
is there no way to achieve the same using math?

Reply:
but it looked cool!